mongo/writeconcern/writeconcern.go

```diff
@@ -17,34 +21,126 @@ import (
 )
 
 // ErrInconsistent indicates that an inconsistent write concern was specified.
+//
+// Deprecated: ErrInconsistent will be removed in Go Driver 2.0.
 var ErrInconsistent = errors.New("a write concern cannot have both w=0 and j=true")
 
 // ErrEmptyWriteConcern indicates that a write concern has no fields set.
+//
+// Deprecated: ErrEmptyWriteConcern will be removed in Go Driver 2.0.
 var ErrEmptyWriteConcern = errors.New("a write concern must have at least one field set")
 
 // ErrNegativeW indicates that a negative integer `w` field was specified.
+//
+// Deprecated: ErrNegativeW will be removed in Go Driver 2.0.
 var ErrNegativeW = errors.New("write concern `w` field cannot be a negative number")
 
 // ErrNegativeWTimeout indicates that a negative WTimeout was specified.
+//
+// Deprecated: ErrNegativeWTimeout will be removed in Go Driver 2.0.
 var ErrNegativeWTimeout = errors.New("write concern `wtimeout` field cannot be negative")
 
-// WriteConcern describes the level of acknowledgement requested from MongoDB for write operations
-// to a standalone mongod or to replica sets or to sharded clusters.
+// A WriteConcern defines a MongoDB read concern, which describes the level of acknowledgment
+// requested from MongoDB for write operations to a standalone mongod, to replica sets, or to
+// sharded clusters.
+//
+// For more information about MongoDB write concerns, see
+// https://www.mongodb.com/docs/manual/reference/write-concern/
 type WriteConcern struct {
-	w interface{}
-	j bool
+	// W requests acknowledgment that the write operation has propagated to a specified number of
+	// mongod instances or to mongod instances with specified tags. It sets the the "w" option
+	// in a MongoDB write concern.
+	//
+	// W must be a type string or int value.
+	//
+	// For more information about the "w" option, see
+	// https://www.mongodb.com/docs/manual/reference/write-concern/#w-option
+	W interface{}
+
+	// Journal requests acknowledgment from MongoDB that the write operation has been written to the
+	// on-disk journal. It sets the "j" option in a MongoDB write concern.
+	//
+	// For more information about the "j" option, see
+	// https://www.mongodb.com/docs/manual/reference/write-concern/#j-option
+	Journal *bool
```

Comment:
What are your thoughts about keeping using `J` as it is described so in the docs?

Reply:
If I understand it correctly, we use a pointer here to make it omittable. However, it seems syntactically odder to me. Do you think allowing a tag or using a tri-valued type helps?

Reply:
After doing some more digging, I found an interesting [Go issue](https://github.com/golang/go/issues/45624) discussing the difficulty of declaring pointers to constants or literals. That issue is a proposal to add a syntax for declaring a pointer to a constant, like `new(true)`, but also includes some suggestions about how you can define your own helpers to do the same thing. For example, if you're using Go 1.18+, you can create a `ptr` helper:
```go
func ptr[T any](v T) *T { return &v }
```
that would allow you to write `ptr(true)`.

I'm going to update this PR with some testable examples showing how users should use the API, including how to use the struct literal syntax.